### PR TITLE
Improve clusterbuster motif parsing

### PR DESCRIPTION
**Pull request by ghuls**
- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

``Bio.motifs.clusterbuster`` now supports parsing GAP and WEIGHT parameters
and can optionally write Cluster Buster motif files with floats instead of
integers, by specifying the ``precision=<int>`` parameter when writing:
  e.g. motifs.write(motifs, "clusterbuster", precision=2)

### CODE REVIEW

File: Bio/motifs/clusterbuster.py [hunk 1]
```diff
@@ -58,21 +70,30 @@ def read(handle):
...
+    motif.gap = motif_gap
+    motif.weight = motif_weight
     record.append(motif)
 
     return record
 
 
-def write(motifs):
+def write(motifs, precision=0):
     """Return the representation of motifs in Cluster Buster position frequency matrix format."""
```
Comment (peterjc) [inline Bio/motifs/clusterbuster.py:81]: Can you briefly describe the precision argument here, with default 0 meaning integers.

Note there should be a blank line following the current one-line summary, and then the new text.

Comment (ghuls) [reply Bio/motifs/clusterbuster.py:81]: done.

**Review by mdehoon — APPROVED** [review]

File: Bio/motifs/clusterbuster.py (hunk 1 above)
Comment (peterjc) [reply Bio/motifs/clusterbuster.py:81]: Did you forget to push a commit?

Comment (ghuls) [reply Bio/motifs/clusterbuster.py]: Yes I forgot. Force pushed now.

Comment (peterjc) [reply Bio/motifs/clusterbuster.py]: The opening triple-quote and the one-line summary should be on the same line
https://peps.python.org/pep-0257/#multi-line-docstrings

That this appears to have passed the automated tests puzzles me... possibly a flake8 to ruff regression? i.e. Perhaps ruff misses this.

Comment (ghuls) [reply Bio/motifs/clusterbuster.py]: > The opening triple-quote and the one-line summary should be on the same line https://peps.python.org/pep-0257/#multi-line-docstrings
> 
> That this appears to have passed the automated tests puzzles me... possibly a flake8 to ruff regression? i.e. Perhaps ruff misses this.

From that link:
> The summary line may be on the same line as the opening quotes **or on the next line**. The entire docstring is indented the same as the quotes at its first line (see example below).

Comment (peterjc) [reply Bio/motifs/clusterbuster.py]: My memory may be at fault, but likely ruff takes a less strict interpretation that the flake8 based system we used to use.

**Comment by peterjc** [comment]
Thank you!

**Pull request merged**